enricher/eks/eks.go

```diff
@@ -50,18 +52,26 @@ func (e Enricher) EnrichRecord(r map[interface{}]interface{}, t time.Time) map[i
 		mappings.RESOURCE_PLATFORM:               e.CloudPlatform,
 		mappings.RESOURCE_PROVIDER:               e.CloudProvider,
 	}
-
 	r[mappings.OBSERVED_TIMESTAMP] = t.UnixMilli()
 
-	// If Fluentbit has failed to enrich k8s metadata on the log, we insert a placeholder value for the kubernetes.service_name
-	// https://docs.google.com/document/d/1vRCUKMeo6ypnAq34iwQN7LtDsXxmlj0aYEfRofwV7A4/edit
+	// If Fluentbit has failed to enrich k8s metadata on the log, this can mean two things:
 	if _, ok := r[mappings.KUBERNETES_RESOURCE_FIELD_NAME]; !ok {
-		r[mappings.KUBERNETES_RESOURCE_FIELD_NAME] = map[interface{}]interface{}{
-			mappings.KUBERNETES_CONTAINER_NAME: mappings.PLACEHOLDER_MISSING_KUBERNETES_METADATA,
+		// The log is a journal systemd log
+		if _, transportOk := r[mappings.TRANSPORT_FIELD_NAME]; msgOk && transportOk {
+			r[mappings.RESOURCE_FIELD_NAME].(map[interface{}]interface{})[mappings.RESOURCE_SERVICE_NAME] = "eks_host_log"
+		} else {
+			// The pod has started up and potentially died too early for us to enrich the log
+			// We insert a placeholder value for the kubernetes.container_name
+			// https://docs.google.com/document/d/1vRCUKMeo6ypnAq34iwQN7LtDsXxmlj0aYEfRofwV7A4/edit
+			r[mappings.KUBERNETES_RESOURCE_FIELD_NAME] = map[interface{}]interface{}{
+				mappings.KUBERNETES_CONTAINER_NAME: mappings.PLACEHOLDER_MISSING_KUBERNETES_METADATA,
+			}
 		}
 	}
 
+	// Add kubernetes static attributes
 	r[mappings.KUBERNETES_RESOURCE_FIELD_NAME].(map[interface{}]interface{})[mappings.KUBERNETES_RESOURCE_CLUSTER_NAME] = e.K8sClusterName
+	r[mappings.KUBERNETES_RESOURCE_FIELD_NAME].(map[interface{}]interface{})[mappings.KUBERNETES_RESOURCE_NODE_NAME] = e.K8sNodeName
```

Comment:
OOC, Does this mean that we can remove this mapping in the EKS standard enricher?

Reply:
Yes, Ill bake this into the EKS base enricher.